### ISSUE TITLE
Loading 

### ISSUE THREAD

**Issue by aschaen** [WiscAr/pychron#4]
On the Loading window: multiple duplicates of different Usernames, Loads, Irradiation, Level, and Labnumber have populated in each drop down menu. Every time something is selected from a drop down, it appears to duplicate itself. 

Can this bug be fixed and the duplicates cleared? Anything I can do to help?

**Comment by aschaen** [comment]
This issue has been mostly solved. There is still a bug: after loading an initial Labnumber and then selecting a new Labnumber, the sample info below does not update automatically, it requires an additional click of the drop down menu. 

After further using the loading window we need the following additional functionality:

1) the weight (mg) input only accepts integers and is a required field. This is troublesome for when we load single or several crystals in one well. We need a way to indicate crystal number e.g. n=1 or n=3 etc. which would overwrite the weight requirement. 

2) in the view menu on the bottom left of this window, it would be useful to display additional parameters like sample #, phase, and crystal number (mentioned above) on the tray layout. having the option to display several of these fields for one well is ideal (i.e. how it works now with labnumber and weights)

3) in the display list on the right side of the widow, it would be useful to be able to edit what is displayed. Right now I can't get it to show the phase, weight etc. Ideally, we would be able to select and deselect any of the fields mentioned above.

**Comment by jirhiker** [comment]
1.  Weight is not required and it accepts any rational number. Use note to indicate number of xtals
2. This is difficult as there is only a limited space for each well. Likely result in a bunch of overlapping text completely obviating the purpose of displaying the information
3. Displaying weight does really make sense for that table. The table groups by L# so a single weight is not relevant.

**Comment by jirhiker** [comment]
Im realizing that Loading will need a significant redesign so the previous comment is not really relevant. 

Some aspects are correct like "Weight is not required and accepts any rational number" and the issue of limited space.  

The redesign should make your requests more readily doable

**Comment by aschaen** [comment]
My initial request might not have been specific enough and the generalizations I suggested may have been misleading. Let me simply what we need.

1) If the "weight" integer input was replaced to a string input called "mass" or "unit" then we would have the option to enter either e.g. "60 mg" or "n=2". Putting the crystals in notes doesn't help us because we need it displayed on the tray. 

2) The option we need most is the ability to display this "unit" info on the tray for both a weight and crystal count. If the above #1 is doable and becomes a string, the space issue shouldn't be impacted. Displaying sample # and phase on the tray is unneeded.

3) If the above #1 and #2 are doable this simplifies what we need on the positions list. I see your point with the grouping and that doesn't need to be modified. We only need the ability to show the phase on this position list which will be identical for each labnumber. The irradiation column is superfluous to us so I thought it could be nice to hide it but this is not absolutely needed either. 

Unsure if the Loading needs a total redesign from scratch for a couple small modifications.

**Comment by jirhiker** [comment]
your requests just elucidated the poor original design. "From scratch" was a bit excessive and shouldn't be construed as meaning this will take a ton of work. I just commonly like to start from scratch instead of trying to shoehorn in new features. I started to try to make the requested modifications but things began to unravel. 

1. Ill probably keep the weight input field and add another field for nxtals
2. no problem once the new field is added
3a. Ill add either two tables or the ability to toggle between grouping and not grouping
3b. Ill add the ability to configure the columns displaced in either table

Im away for the rest of the week so thats really the main reason ill said it will take some time. 

I could probably make the first 2 items work in the current structure without a huge amount of work but its better to do it right from the beginning. I designed Loading in the abstract without ever having any users use it so that now that you are using it all the early design problems are getting flushed out. This is good. So dont hesitate to ask for features.
